### PR TITLE
WT-6908 Write "cache" subpage for Architecture Guide

### PR DESCRIPTION
**Pull request by keitharnoldsmith**
Draft of cache page for Architecture Guide.

The boundaries of what belongs here as opposed to related pages in the Arch Guide was fuzzy. I debated including a description of read generations here, but decided that would be a better fit on the eviction page. Let me know if there are more topics or details that you think belong here.

### CODE REVIEW

File: src/docs/arch-cache.dox [hunk 1]
```diff
@@ -1,13 +1,110 @@
 /*! @arch_page arch-cache Cache
 
-Cache in WiredTiger is represented by the various shared data structures
-that make up in-memory Btrees and subordinate data structures.
+The WiredTiger cache is memory used to hold copies of recently accessed or modified data.
+WiredTiger reads btree pages into the cache on demand.  When the cache runs low on space, Eviction
```
Comment (ddanderson) [inline src/docs/arch-cache.dox:4]: In the arch guide, we've been trying to use "Btree" instead of "btree".

File: src/docs/arch-cache.dox [hunk 2]
```diff
@@ -1,13 +1,110 @@
...
+A \c WT_REF can represent either
+a page in the cache or one that has not been loaded yet.
+The page itself is represented by a \c WT_PAGE structure.  This includes a pointer to a buffer
+that contains the on-disk page image (decrypted and uncompressed).  It also holds the supplemental
+structures that WiredTiger uses to access and update the page while it is cached.
+
+When WiredTiger loads a page into the cache, it allocates an internal table with one entry
+for each entry on the page.  The type of these entries depends on the page type.  An internal
+btree page will have an array of \c WT_REF structures.  A btree leaf page will have an array
+of \c WT_ROW structures representing the KV pairs.  WiredTiger uses this table to find entries
```
Comment (ddanderson) [inline src/docs/arch-cache.dox:56]: Could be WT_COL structures as well?

**Review by ddanderson — APPROVED** [review]
This looks good, very readable.  I can a couple of small comments.

File: src/docs/arch-cache.dox (hunk 1 above)
Comment (keitharnoldsmith) [reply src/docs/arch-cache.dox]: Done.

File: src/docs/arch-cache.dox (hunk 2 above)
Comment (keitharnoldsmith) [reply src/docs/arch-cache.dox]: Yeah.  I was simplifying a bit by skipping the column store details.  I've added a few sentences to describe the fix- and variable-length column store options.

**Pull request merged**